### PR TITLE
feat(game): Connect Frontend to Backend via API Gateway (#80)

### LINKED ISSUE #80: 🎯 MILESTONE: End-to-End Integration via API Gateway (Sprint Week 1)
## 🎯 Sprint Goal

**"Users can login, view their profile, create/join a game through the gateway, and play Pong online"**

---

### ✅ What We Have (Infrastructure 80% Complete)
- **Backend Services:** Game-service HTTP + Socket.IO ready; Auth service with JWT/2FA
- **API Gateway:** HTTP routes wired; WebSocket proxy at `/api/games/ws`
- **Frontend Core:** HttpClient with interceptors, Component system, Signal state management
- **UI Components:** GameLobby, Canvas renderer, Profile components (isolated)
- **Architecture:** Clean patterns established (services layer, Component lifecycle)

### ⚠️ The Gap: Integration Debt
```
Backend Services       ✅ [Game API, Auth API, User API]
                       ↕️ ??? (disconnected)
API Gateway            ✅ [HTTP proxy, WS proxy]
                       ↕️ ??? (frontend not using it)
Frontend Infrastructure ✅ [HttpClient, WebSocket factory]
                       ↕️ ??? (not wired to UI)
Frontend UI            🟡 [Lobby, Canvas, Profile] (isolated, using mocks)
```

**The Problem:** We have components but no connected user journey. No one can play the game yet.

---

## 🚀 Sprint Strategy: Connect Before Create

**Stop:** Building new isolated features  
**Start:** Connecting existing pieces into working flows

### Product-Focused Approach
❌ Wrong: "Build WebSocket client, game canvas, profile page"  
✅ Right: "Users can complete full auth → profile → game creation → play flow"

---

## 📋 Sub-Issues Breakdown (4 Assignable Tasks)

This milestone is broken into **4 parallel tracks** that converge into one working system:

### 🔴 Critical Path (Must Complete First)
1. **Issue #69:** Auth + Profile Integration (Frontend) → **Unblocks everything**
2. **Issue #70:** Online Game Flow via Gateway (Frontend + Backend) → **Core value**

### 🟡 Supporting (Can Start in Parallel)
3. **Issue #71:** User → Game Service Event Bus (Backend) → **Data consistency**
4. **Issue #72:** API Docs + Environment Alignment (DevOps) → **Team clarity**

---

### Issue #69: Auth + Profile Integration (Frontend)
**Owner:** @yabdoul    
**Pair with:** @sh1hvb  (user-service owner)  
**Time:** 2 days

**Deliverable:** Login → Dashboard → Profile (fully connected)

**Key Tasks:**
- Hook login/register to `/api/auth/*` via HttpClient
- Persist JWT in `appState.auth` Signal + localStorage
- Fetch `/api/users/me` on app load → populate profile page
- Add logout flow (clear tokens, redirect to login)

**Demo:** User logs in → sees dashboard → clicks profile → sees real data

---

### Issue #70: Online Game Flow via Gateway (Frontend + Backend)
**Owner:** @abouguri   
**Pair with:** @NourMellal (game-service owner)  
**Time:** 2 days

**Deliverable:** Create game → Join lobby → Play online (via gateway)

**Key Tasks:**
- Wire GameLobby to `/api/games` via HttpClient (through gateway)
- Connect WebSocket using `VITE_WS_GAME_PATH=/api/games/ws/socket.io`
- Subscribe to `game:state_update` → render in Canvas
- Handle paddle input → send via WS → server updates → canvas updates
- Keep local arcade mode behind feature flag

**Demo:** User clicks "Create Game" → game created via gateway → opponent joins → ball moves

---

### Issue #71: User-Service → Game-Service Events (Backend)
**Owner:** @sh1hvb   
**Time:** 1 day

**Deliverable:** User deletion triggers game cleanup

**Key Tasks:**
- Emit `user-deleted` event from user-service to event bus
- Add integration test proving game-service receives event
- Verify game-service cancels active games for deleted user
- Document event contract in `docs/events/user-deleted.md`

**Demo:** Delete user → active games cancelled → opponent sees "Player disconnected"

---

### Issue #72: API Docs + Environment Config Alignment (DevOps)
**Owner:** @NourMellal  
**Time:** 1 day (Wed)

**Deliverable:** Docs match reality, no magic localhost ports

**Key Tasks:**
- Update `docs/api/paths/games.yaml` to match gateway payloads
- Add `VITE_WS_GAME_URL` / `VITE_WS_GAME_PATH` to `.env.example`
- Document gateway WS usage in `docs/development/websocket-setup.md`
- Verify Vault secrets include gateway API keys

**Demo:** New dev clones repo → reads docs → connects to gateway (no guessing)

---

### Day 1-2: Integration Sprint
**Focus:** Connect existing pieces
- **Pair 1:** @yabdoul  + @sh1hvb  → Auth integration (Issue #69)
- **Pair 2:** @abouguri  + @NourMellal → Game flow (Issue #70)
- **Demo:** Mid-sprint check-in EOD (show progress in Discord)

### Day 3: Backend + Docs
- @sh1hvb  → User events (Issue #71)
- @NourMellal → Docs alignment (Issue #72)
- **Integration test:** Auth + Game flow working together

### Day 4: Polish + Bug Bash
- Fix critical bugs from integration testing
- Error handling (network failures, 401s, disconnects)
- Loading states everywhere
- **Goal:** Everything works end-to-end

### Day 5: Demo Prep + Retrospective
**Morning:** Record demo video
- Login flow
- Profile view
- Game creation via gateway
- Live Pong gameplay

**Afternoon:** Sprint retrospective
- What shipped?
- What blocked us?
- What to improve next sprint?

---

### Technical Acceptance Criteria
- [ ] User can login via `/api/auth/login` (gateway)
- [ ] Dashboard shows profile from `/api/users/me` (gateway)
- [ ] User can create game via `/api/games` (gateway)
- [ ] WebSocket connects via `/api/games/ws/socket.io` (gateway proxy)
- [ ] Canvas renders real-time game state from WebSocket
- [ ] Paddle input sent via WS → server → opponent sees it
- [ ] Deleting user cancels their active games (backend event)
- [ ] `.env.example` has all required variables documented
- [ ] API docs match actual payloads

### Product Acceptance Criteria (Demo-able)
- [ ] Can record 5-minute video showing full flow
- [ ] No "TODO" or "mock data" in critical paths
- [ ] Error states handled (network failure, 401, disconnects)
- [ ] Works on Chrome + Firefox (basic browser testing)

---

### ❌ "Merge Now, Fix Later"
**Problem:** PR approved with "still needs enhancements" (PR #73 example)  
**Fix:** If not production-ready, don't merge. Fix it now or remove feature.

### ❌ "Works on My Machine"
**Problem:** Features work isolated but break when integrated  
**Fix:** Integration test on Day 3 (mid-sprint check)

### ❌ Silent Blockers
**Problem:** Dev stuck for 2 days, doesn't mention in standup  
**Fix:** Post blockers in Discord within 2 hours. Tag lead immediately.

### ❌ Component-Focused PRs
**Problem:** PR titled "Add game component" (no user context)  
**Fix:** Every PR must have user story: "As a user, I want to [X] so that [Y]"

---

## 🔗 Sub-Issue Links

- #69 - Auth + Profile Integration (Frontend)
- #70 - Online Game Flow via Gateway (Frontend + Backend)  
- #71 - User → Game Service Events (Backend)
- #72 - API Docs + Environment Alignment (DevOps)

---

## 📚 Resources

- **Integration Milestone Doc:** `docs/development/milestone-integration.md`
- **API Gateway Config:** `services/api-gateway/src/config/`
- **WebSocket Client:** `apps/web/src/modules/shared/services/WebSocketClient.ts`
- **Environment Variables:** `.env.example` (must be updated in Issue #72)

---

### Blocker Resolution SLA
- **Post blocker in Discord within 2 hours of getting stuck**
- **Lead responds within 1 hour with unblock plan**
- **Pair programming session scheduled same day if needed**

### Pull Request Template
Every PR must include:
```markdown
## User Story
As a [user], I want to [goal] so that [benefit]

## Demo
[Screenshot or GIF showing feature working]

## Changes
- Bullet list of technical changes

## Test Plan
1. Step-by-step manual test
2. Edge cases tested

## Breaking Changes
None / [describe]
```

---

### Quantitative
- [ ] 4 sub-issues closed by Friday
- [ ] 0 critical bugs in production paths
- [ ] 100% of acceptance criteria met

### Qualitative
- [ ] Can demo to unfamiliar person without explaining code
- [ ] Each feature connects to backend (no mocks in demo)
- [ ] Team knows next sprint goal before this sprint ends

---

## 🏁 Next Sprint Preview (Week 2)

**Assuming Sprint 1 succeeds:**
- Real-time tournament bracket creation
- Friend invite to game
- Match history + stats tracking
- Mobile optimization

**Goal:** By end of Week 2, users can play tournaments with friends.

---

## 💬 Questions or Concerns?

**Post in Discord #dev-questions or tag @NourMellal**

This is our **integration sprint**. Focus = connect, not create.

Let's ship something users can actually use! 🚀

### PR DESCRIPTION
**Pull request by abouguri**
## Connect Frontend Game to Backend via API Gateway

Closes #80

### Overview

Implements frontend-backend integration for real-time multiplayer Pong. Connects GameCanvas and GameLobby components to game-service via API Gateway using HTTP and WebSocket protocols. Includes feature flag for local arcade mode.

### Changes

**New Components:**
- `GameService.ts` - HTTP client for game CRUD (/api/games endpoints)
- `gameSocket.ts` - Socket.IO client with JWT auth and reconnection logic
- `featureFlags.ts` - Environment-based mode switching (local/online)
- `CreateGamePage.ts` - Game creation interface

**Modified Components:**
- `GameCanvas.ts` - WebSocket integration, event delegation pattern
- `GameLobby.ts` - Real-time updates via WebSocket events
- `game.types.ts` - Type definitions aligned with OpenAPI spec
- `.env.example` - Configuration for WebSocket endpoints

### Architecture

- HTTP requests route through API Gateway (`/api/games`)
- WebSocket connects via Gateway proxy (`/api/games/ws/socket.io`)
- JWT authentication in query parameters
- Feature flag: `VITE_GAME_MODE=local` (offline) or `online` (server-authoritative)
- Clean separation between domain types and infrastructure

### Testing

**Local Mode:** Verified functional (ball physics, paddle controls, scoring)
**Online Mode:** Implementation complete, requires backend services for integration testing

### Related

Issue #80, #67, #65 | PR #75, #61

### CODE REVIEW

File: infrastructure/api-gateway/src/config/gateway-config.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ const DEFAULT_SERVICE_URLS = {
     tournament: 'http://localhost:3005'
 } as const;
 
-const DEFAULT_CORS = ['http://localhost:3000'];``
+const DEFAULT_CORS = ['http://localhost:3000', 'http://localhost:3002'];
```
Comment (sh1hvb) [inline infrastructure/api-gateway/src/config/gateway-config.ts]: i suggest to give 2999 to gateway cause web runnig at 3000 and chat at 3002

**Review by NourMellal — APPROVED** [review]
i think were good with this util here buddy @abouguri we can merge it! And big thanks for your work
